go/vt/key/key.go

```diff
@@ -26,16 +26,19 @@ import (
 	"regexp"
 	"strings"
 
-	"google.golang.org/protobuf/proto"
-
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 )
 
+var (
+	ZeroKey         = []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+	KeyRangePattern = regexp.MustCompile(`^[0-9a-fA-F]*-[0-9a-fA-F]*$`)
```

Comment:
I agree, I only relocated the existing regexp constant from where it was placed far below in the same file but it does need work; in fact it also needs to only accept multiples of two digits. I would propose leaving fixing this up to a follow-up PR though, so that the actual fix and associated tests can be much more clear. I'll be happy to create an issue for that and even work on it though.

Reply:
Resolving this since now I know it's of variable length.